### ISSUE TITLE
Emoticon standard

### ISSUE THREAD

**Issue by 2Cubed** [CactusDev/CactusBot#55]
We need a standard for defining emoticons in `MessagePacket`s (#43).

---

For Unicode emoticons, we should follow [emoji-alpha-codes](https://github.com/Ranks/emoji-alpha-codes). So, :stuck_out_tongue: should be saved with the data `:stuck_out_tongue:`.

However, there are lots of emoticons which do not appear in Unicode. (For example, `:mappa`/`Kappa`.) What do we do in this case? Also, because we can only use each name once, where do we store the list of used names? (Preferably with descriptions of what each means.)

**Comment by Innectic** [comment]
Well, since `:mappa` and `Kappa` are used to show sarcasm, maybe we just say `sarcasm` in the file?

**Comment by 2Cubed** [comment]
@Innectic I agree. Currently, EAC's are being stored with the `::`s - I guess custom ones can be stored without.

However, the question still stands... where do we store our global list of custom codes? We need to specify somewhere what each custom code means, and keep track of every one we define so that we don't accidentally duplicate any and make the conversions not work.

**Comment by Innectic** [comment]
We already took care of this, didn't we?

**Comment by 2Cubed** [comment]
@Innectic Not entirely. I'm OK with storing Unicode emojis with the EAC alphas and the non-Unicode ones with non-`::`s.

Where do we put the global list of non-Unicode emojis, though? We need a "table" of sorts that details every custom code we're using, and what it defines. (If we don't have this, it would be easy to forget that, say, `:mappa` is `sarcasm` and use `troll` for `Kappa`.) Would a giant JSON file, similar to the EAC, work? We could put it in `docs/`, maybe, and have a documentation file that uses JS to load it all into a table? :thinking:

**Comment by Innectic** [comment]
We have a json file in the base of the `beam` service that handles that. I thought that's what we were doing for everything?

**Comment by 2Cubed** [comment]
@Innectic Yes, we need those. However, those are just mappings for each service. (`:cactus` -> `:cactus:`, `:)` -> `:smiley:`, `:mappa` -> `sarcasm`, etc.) We need a global file that explains what every custom code (like `sarcasm`) means, or at least lists them.

```json
{
  "sarcasm": "Implies that the statement is sarcastic.",
  "salute": "Face with saluting hand."
}
```

**Comment by 2Cubed** [comment]
Should we use Unicode as our default instead of Emoji Alpha Codes? This would simplify the translation for services like Discord and make cross-service displaying look (arguably) better overall.

**Comment by Innectic** [comment]
@2Cubed Beam chat doesn't support it and neither does twitch. That means only one planned platform can support it

**Comment by 2Cubed** [comment]
@Innectic Unicode is the universal standard. We're going to be translating to and from it - we need not worry about what services support it natively. (And even if we were worrying about that, Unicode emoji render *as actual emoji* on Beam and Twitch, whereas `:stuck_out_tongue:` won't.)

**Comment by Innectic** [comment]
That's the point of the annoying file called `emoji.json`. Converting emotes.

**Comment by 2Cubed** [comment]
@Innectic Yes. The question is, should `:)` be translated to `:smile:`, or to `🙂`? This decision would only be seen if we don't have a translation for a specific emoji, but we should pick a good standard nonetheless.

I think we should use Unicode when it's a Unicode emote (`🙂`), but EAC format when it's a custom emote (`:sarcasm:`, `:python:`, etc.).

**Comment by 2Cubed** [comment]
Done in #139.
